### PR TITLE
Rolling restart e2e test

### PR DESCRIPTION
**Pull request by udnay**
Testing the rolling restart functionality with e2e test. Validates that
the pods at the end of the test all have a newer timestamp than when the
cluster was initialized.

### CODE REVIEW

File: e2e/e2e_test.go [hunk 1]
```diff
@@ -368,3 +372,100 @@ func TestPVCResize(t *testing.T) {
...
+
+	testLog := zapr.NewLogger(zaptest.NewLogger(t))
+
+	actor.Log = testLog
+
+	sb := testenv.NewDiffingSandbox(t, env)
+	sb.StartManager(t, controller.InitClusterReconcilerWithLogger(testLog))
+
+	builder := testutil.NewBuilder("crdb").WithNodeCount(3).WithTLS().
+		WithImage("cockroachdb/cockroach:v20.2.5").
```
Comment (keith-mcclellan) [inline e2e/e2e_test.go:394]: We need an enhancement to the testing framework that pulls this from a central place we can keep up to date.  There are relatively few tests where we should be pinning to specific releases and those are largely around upgrades

File: e2e/e2e_test.go [hunk 2]
```diff
@@ -368,3 +372,100 @@ func TestPVCResize(t *testing.T) {
...
+					} else if rt == "" { // The annotation was deleted
+						return true, nil
+					}
+
+					return false, nil
+				})
+				require.NoError(t, err)
+
+				// Make sure the cluster is in a good state after the restart is finished
+				RequireClusterToBeReadyEventuallyTimeout(t, sb, builder, 500*time.Second)
```
Comment (keith-mcclellan) [inline e2e/e2e_test.go:445]: I think your timeout here will be too short if we merge the safer restart code Alina is working on.  This is a great reason why we need to do a better job parallelizing these test cases.

File: e2e/e2e_test.go [hunk 3]
```diff
@@ -368,3 +372,100 @@ func TestPVCResize(t *testing.T) {
...
+						continue // Skip the validate version pod
+					}
+					podCreationTimes[pod.Name] = pod.CreationTimestamp
+				}
+
+				current.Annotations["crdb.io/restarttype"] = "rolling"
+				require.NoError(t, sb.Update(current))
+
+				// Need to poll for getting rid of the restart annotation
+				err = wait.Poll(10*time.Second, 500*time.Second, func() (done bool, err error) {
```
Comment (keith-mcclellan) [inline e2e/e2e_test.go:428]: I think this is likely to fail once we add the enhanced restart logic because of a timeout issue

**Review by keith-mcclellan — CHANGES_REQUESTED** [review]
There could be other items but I noticed a few spots where I'd expect this to fall over, particularly after we merge Alina's enhanced health check PR

**Comment by udnay** [comment]
Going to close this as we test restart a lot implicitly already and our e2e tests need some rethinking

**Pull request closed without merging**